### ISSUE TITLE
[Feature request] Accessible Firewall Usage Guides

### ISSUE THREAD

**Issue by issuant** [mullvad/mullvadvpn-app#7072]
### I have checked if others have suggested this already

- [X] I have checked this issue tracker to see if others have reported similar issues.

### Feature description

Since an integrated firewall is unfortunately currently out of scope (https://github.com/mullvad/mullvadvpn-app/issues/6775#issuecomment-2410116849), can we instead get accessible, beginner-friendly guides on how to use the VPN with the default Windows firewall, and also (non-root) Android firewalls? Of the unspecific desktop firewall guides that do exist out there, most are either written for Linux and derivatives, those with domain knowledge, breakdown once you turn on Mullvad, recommend or are about apps that do not work simultaneously with Mullvad, or have the potential to go very wrong if configured improperly.

### Alternative solutions

n/a

### Type of feature

- [ ] Better privacy/anonymity
- [ ] Better at circumventing censorship
- [ ] Easier to use
- [X] Other

### Operating System

- [X] Android
- [ ] iOS
- [X] Windows
- [ ] macOS
- [ ] Linux

**Comment by Serock3** [comment]
Hello. The VPN app automatically sets up firewall rules on all desktop platforms. You shouldn't have to configure anything manually unless you are hosting a server, which is not very common on windows, or want to poke very specific holes on the firewall in a way that is not covered by our split tunneling functionality. Is there any specific issue you are having the with the windows firewall?

Non-rooted Android devices do not have access to firewalls in the same sense as desktops. As far as I know, any third party firewall works by hooking into the VPN API provided by Android, and it only supports one VPN at a time.

**Comment by Serock3** [comment]
Sorry for being slow to reply. I tested applying a l simple firewall rule in windows to block a single app, and it works fine regardless of whether you are connected in the VPN or not. Here's a beginner friendly guide for how to do it <https://techcommunity.microsoft.com/t5/windows-11/how-to-block-a-program-from-accessing-the-internet-windows-11/m-p/2772951>. Could you try that and tell me if it works for you? If our VPN app doesn't affect the way to set up firewall rules then it's probably not relevant for us to write out own guide.

@kl, could you elaborate on the Android side of the issue?

**Comment by kl** [comment]
On Android there are firewall apps such as NetGuard. However, such apps will not work when using the Mullvad app because they also need to set up their own VPN tunnel in order to block specific app traffic, and we can't have multiple VPN apps running at the same time on Android.  So it won't be possible to block specific apps unless a firewall feature is added to the Mullvad app.

**Comment by Serock3** [comment]
It's probably an issue specific to VSCode. I get the same behavior as you for it, but not for other apps. It may be worth mentioning that traffic coming from WSL will not be blocked by the normal Windows firewall, in case you are running apps from there.

**Comment by Serock3** [comment]
Regarding Android. It's a shame that it doesn't support blocking internet access for individual apps natively. If you are really interested in that feature you could look into [GrapheneOS](https://grapheneos.org/#grapheneos). From my discussions with the Android team it seems that implementing firewall functionality, while still maintaining a functioning VPN tunnel, is not as easy as it seems, unfortunately.

**Comment by Serock3** [comment]
The reason Windows firewall isn't able to block some apps by their executable path is likely due to them spawning sub-processes that actually handle their network traffic. This is not related to the VPN service and unfortunately not easy to get around using Windows Defender, as far as I am aware.

I should mention that the split tunneling driver on Windows can handle these problematic apps. This is due to it being a custom made kernel module which dynamically captures and re-routes sub-processes as they are created. It would be great if Windows Defender implemented something similar. As it stands there is really nothing we could write a guide about, unfortunately.

**Comment by Serock3** [comment]
>    >  I should mention that the split tunneling driver on Windows can handle these problematic apps
> 
> What does this mean?

Our split tunneling driver listens for when apps spawn sub-processes. If you split tunnel an app that spawns a new process to handle network traffic, the traffic of the new process will be routed outside the tunnel as well.

I don't have any recommendations on third party firewalls, unfortunately. I hope you will find one that suits your needs. If there are any Windows firewalls that can block apps with sub-processes, then my guess is that they will work fine regardless of whether the VPN is active or not. I'm sorry to say it, but it's outside the scope of our VPN guides to instruct on how to set up a third party firewalls. I will close this issue now.
